### DISCUSSION TITLE
field的datasource初始化的问题

### DISCUSSION THREAD

**Discussion by dingchaoyan1983** [alibaba/formily#3909] [Q&A 2.x]
我想将Filed和useQuery集成起来，但是useQuery只能放在组件的顶层，通过useQuery 返回的数据，怎么初始化 filed的datasource
```
const { data } = useQuery({ queryKey: [], queryFn })

return <Field  reactions={....}/>
```

请问怎么去初始化这个datasrouce，从哪里传入？
因为useQuery是个hook不能写在reactions中只能写在外层

**Comment by dingchaoyan1983** [comment]
好像只能在Field的component里面去用这个hook，然后用useEffect去修改filed的datasource属性

**Comment by Arsenalfanscz** [comment]
请问下遇到类似的问题有解决办法了吗
